src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -191,4 +195,333 @@ export class CircleEditorDocument extends Disposable implements vscode.CustomDoc
+    if (name === undefined || name === undefined || name === undefined || name === undefined) {
+      Balloon.error("input data is undefined", false);
+      return;
+    }
+    if (argument._initializer !== null) {
+      const ini = argument._initializer;
+      if (isChanged === true) {
+        bufferData = ini._data;
+      }
+      isVariable = ini._is_variable;
+    }
+    tensorType = tensorType.toUpperCase();
+
+    const targetTensor = this._model?.subgraphs[subgraphIdx]?.tensors[tensorIdx];
+    if (targetTensor === undefined) {
+      Balloon.error("model is undefined", false);
+      return;
+    }
+    targetTensor.name = argname;
+    if (tensorType === 'BOOLEAN') {
+      tensorType = 'BOOL';
+    }
+    let tensorTypeNum: any = Circle.TensorType[tensorType];
+    targetTensor.type = tensorTypeNum;
+    targetTensor.shape = tensorShape;
+    if (bufferData !== null) {
+      const editBufferIdx: number = targetTensor.buffer;
+      this._model.buffers[editBufferIdx].data = bufferData;
+    }
+    return;
+  }
+
+  /**
+   * This function edit Attribute about built-in operators and custom operators.
+   * if operator code is 32, encode key and value about custom operators. and overwrite customOptions's int array.
+   * else, edit built-in operator's attributes according to Custom Type and Normal Type in CircleType.ts.
+   */
+  private editAttribute(data: any) {
+    let subgraphIdx: number = Number(data._subgraphIdx);
+    let operatorIdx: number = Number(data._nodeIdx);
+    let inputTypeName: string = data.name;
+    if (inputTypeName === undefined || subgraphIdx === undefined || operatorIdx === undefined) {
+      Balloon.error("input data is undefined", false);
+      return;
+    }
+    inputTypeName = inputTypeName.toUpperCase();
+    let inputTypeOptionName: any = inputTypeName + "OPTIONS";
+    let operatorCode: number = 0;
+    for (let i = -4; i <= 146; i++) {
+      let builtinOperatorKey = Circle.BuiltinOperator[i];
+      if (builtinOperatorKey === undefined) { continue; }
+      let tempKey: any = Circle.BuiltinOperator[i];
+      while (1) {
+        const tempKey2 = tempKey.replace('_', '');
+        if (tempKey.length === tempKey2.length) { break; }
+        tempKey = tempKey2;
+      }
+      builtinOperatorKey = tempKey;
+      builtinOperatorKey = builtinOperatorKey.toUpperCase();
+      if (builtinOperatorKey === inputTypeName) {
+        operatorCode = i;
+        break;
+      }
+    }
+    const operator: any = this._model.subgraphs[subgraphIdx].operators[operatorIdx];
+    if (operator === undefined) {
+      Balloon.error("model is undefined", false);
+      return;
+    }
+    // built-in operator Case
+    if (operatorCode !== 32) {
+      if (operator.builtinOptions === null) {
+        Balloon.error("built-in Options is null", false);
+        return;
+      }
+      if (inputTypeOptionName.indexOf("POOL2D") !== -1) {
+        inputTypeOptionName = "POOL2DOPTIONS";
+      }
```

Comment:
Why only this is handled specially?

Reply:
The built-in operator includes average_pool_2d, L2_pool_2d, and max_pool_2d, all of which have been made exceptions by referring to Pool2D Options.